tests/Collection/CollectionFunctionalTest.php

```diff
@@ -800,6 +802,32 @@ public function testMethodInTransactionWithReadConcernOption($method): void
         }
     }
 
+    public function testListSearchIndexesInheritTypeMap(): void
+    {
+        $this->skipIfAtlasSearchIndexIsNotSupported();
+
+        $collection = new Collection($this->manager, $this->getDatabaseName(), $this->getCollectionName(), ['typeMap' => ['root' => 'array']]);
+
+        // Insert a document to create the collection
+        $collection->insertOne(['_id' => 1]);
+
+        try {
+            $collection->createSearchIndex(['mappings' => ['dynamic' => false]], ['name' => 'test-search-index']);
```

Comment:
Does `dynamic: false` make sense here? Quoting [`createSearchIndex` command docs](https://www.mongodb.com/docs/manual/reference/command/createSearchIndexes/):

> If set to false, you must specify individual fields to index using `mappings.fields`.

Presumably, this creates a search index that does nothing. I'm curious as to why the server would even allow that, but wonder if future validation might break this. `dynamic: true` seems more correct (if only as a valid demonstration).

Reply:
I used this mapping because it is in the spec test: https://github.com/mongodb/specifications/blob/master/source/index-management/tests/README.md#case-1-driver-can-successfully-create-and-list-search-indexes

This requires a spec change if we want a client error in case of `dynamic: false AND fields: empty`

Reply:
Thanks for digging that up. Since spec test reads the definition back from the server, I think this goes beyond anything drivers should address themselves. I'll share a link to this thread with the Atlas Search team but OK to leave this PR as-is.